cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/BudgetsPage.java

```diff
@@ -0,0 +1,324 @@
+    private WebElement editSpendingBudgetButton;
+
+    @FindBy(id = "totalBudgetAmount")
+    private WebElement totalBudgetAmountInput;
+
+    @FindBy(id = "Submit-Spending-Budget")
+    private WebElement submitSpendingBudgetButton;
+
+    @FindBy(id = "Spending-Budget-Div")
+    private WebElement spendingBudgetDiv;
+
+    @FindBy(css = "div.nav-container.bg-\\[\\#005ea2\\].px-6.h-screen.min-w-64.max-w-min-w-64.flex.flex-col.pt-6")
+    private Navbar navbar;
+
+    // Edit and Delete Budget Modal Elements
+    private WebElement editBudgetButton;
+    private WebElement deleteBudgetButton;
+    private WebElement editedBudgetedInput;
+    private WebElement saveBudgetButton;
+    private WebElement confirmDeleteButton;
+    // Edit and Delete Savings Bucket Modal Elements
+    private WebElement editSavingsBucketButton;
+    private WebElement deleteSavingsBucketButton;
+    private WebElement editedAmountRequiredInput;
+    private WebElement saveSavingsBucketButton;
+    private WebElement confirmDeleteSavingsBucketButton;
+
+    private Map<String, WebElement> elementMap;
+    private List<WebElement> buttonList = new ArrayList<>();
+
+    public BudgetsPage(WebDriver driver) {
+        super(driver);
+        PageFactory.initElements(driver, this);
+        elementMap = new HashMap<>();
+        elementMap.put("summaryComponentDiv", summaryComponentDiv);
+        elementMap.put("addNewBudgetButton", addNewBudgetButton);
+        elementMap.put("budgetedInput", budgetedInput);
+        elementMap.put("budgetsTable", budgetsTable);
+        elementMap.put("savingsBucketTable", savingsBucketTable);
+        elementMap.put("budgetNameInput", budgetNameInput);
+        elementMap.put("amountRequiredInput", amountRequiredInput);
+        elementMap.put("submitNewSavingsBucketButton", submitNewSavingsBucketButton);
+        elementMap.put("editBudgetButton", editBudgetButton);
+        elementMap.put("deleteBudgetButton", deleteBudgetButton);
+        elementMap.put("editedBudgetedInput", editedBudgetedInput);
+        elementMap.put("saveBudgetButton", saveBudgetButton);
+        elementMap.put("confirmDeleteButton", confirmDeleteButton);
+        elementMap.put("editSavingsBucketButton", editSavingsBucketButton);
+        elementMap.put("deleteSavingsBucketButton", deleteSavingsBucketButton);
+        elementMap.put("editedAmountRequiredInput", editedAmountRequiredInput);
+        elementMap.put("saveSavingsBucketButton", saveSavingsBucketButton);
+        elementMap.put("confirmDeleteSavingsBucketButton", confirmDeleteSavingsBucketButton);
+        elementMap.put("editSpendingBudgetButton", editSpendingBudgetButton);
+        elementMap.put("totalBudgetAmountInput", totalBudgetAmountInput);
+        elementMap.put("submitSpendingBudgetButton", submitSpendingBudgetButton);
+        elementMap.put("spendingBudgetDiv", spendingBudgetDiv);
+    }
+
+    @Override
+    public List<Component> getChildComponents() {
```

Comment:
Low impact: Redundant code creation of components variable followed by a new ArrayList for returning.
Suggestion: Return components instead.